src/test/java/com/google/sps/servlets/ContestServletTest.java

```diff
@@ -0,0 +1,178 @@
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class ContestServletTest {
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig(), new LocalURLFetchServiceTestConfig());
+
+  @Mock HttpServletRequest httpServletRequest;
+  @Mock HttpServletResponse httpServletResponse;
+  @Mock PrintWriter printWriter;
+
+  private static final ImmutableSet<Long> CANDIDATE_ID_SET =
+      new ImmutableSet.Builder<Long>().add(1L).add(2L).add(3L).build();
+
+  private Entity electionEntityOne;
+  private Entity contestEntityOne;
+  private Entity contestEntityTwo;
+
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    electionEntityOne = new Entity("Election");
+    electionEntityOne.setProperty("id", "9999");
+    electionEntityOne.setProperty("name", "myElection");
+    electionEntityOne.setProperty("scope", "myScope");
+    electionEntityOne.setProperty("date", "myDate");
+    electionEntityOne.setProperty("contests", new HashSet<Long>());
+    electionEntityOne.setProperty("propositions", new HashSet<Long>());
+
+    contestEntityOne = new Entity("Contest");
+    contestEntityOne.setProperty("name", "myFirstContest");
+    contestEntityOne.setProperty("candidates", CANDIDATE_ID_SET);
+    contestEntityOne.setProperty("description", "This contest is important.");
+
+    contestEntityTwo = new Entity("Contest");
+    contestEntityTwo.setProperty("name", "mySecondContest");
+    contestEntityTwo.setProperty("candidates", new HashSet<Long>());
+    contestEntityTwo.setProperty("description", "This contest is also important.");
+  }
+
+  @Test
+  public void singleElection_singleContest_testDoGet() throws Exception {
+    Entity electionEntity = new Entity("Election");
+    Entity contestEntity = new Entity("Contest");
+
+    electionEntity.setPropertiesFrom(electionEntityOne);
+    contestEntity.setPropertiesFrom(contestEntityOne);
+
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    long contestId = ds.put(contestEntity).getId();
+
+    Collection<Long> contestSet = (Collection<Long>) electionEntity.getProperty("contests");
+    contestSet.add(contestId);
+    ds.put(electionEntity);
+
+    when(httpServletRequest.getParameter("electionId")).thenReturn("9999");
+    when(httpServletResponse.getWriter()).thenReturn(printWriter);
+
+    ContestsServlet contestServlet = new ContestsServlet();
+    contestServlet.doGet(httpServletRequest, httpServletResponse);
+
+    verify(printWriter)
+        .println(
+            "[{\"name\":\"myFirstContest\",\"candidates\":[1,2,3],\"description\":\"This contest is important.\"}]");
+  }
+
+  @Test
+  public void singleElection_twoContest_testDoGet() throws Exception {
+    Entity electionEntity = new Entity("Election");
+    Entity firstContestEntity = new Entity("Contest", 1);
```

Comment:
Do we need to specify the id?

Reply:
That was leftover from when we had the setUpClass method but it seems to work fine now without specifying the Id. Will update.

Reply:
Actually still need to specify the Id when testing when a contest isn't present in datastore for the entity that is created but isn't put into datastore.

Reply:
done